src/main/java/com/google/sps/data/FirestoreDB.java

```diff
@@ -0,0 +1,265 @@
+package com.google.sps.meltingpot.data;
+
+import com.google.api.core.ApiFuture;
+import com.google.cloud.firestore.CollectionReference;
+import com.google.cloud.firestore.DocumentReference;
+import com.google.cloud.firestore.DocumentSnapshot;
+import com.google.cloud.firestore.FieldValue;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.GeoPoint;
+import com.google.cloud.firestore.Query;
+import com.google.cloud.firestore.QueryDocumentSnapshot;
+import com.google.cloud.firestore.QuerySnapshot;
+import com.google.cloud.firestore.WriteResult;
+import com.google.firebase.cloud.FirestoreClient;
+import com.google.sps.meltingpot.data.DBUtils;
+import com.google.sps.meltingpot.data.User;
+import java.lang.Iterable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+public class FirestoreDB implements DBInterface {
+  public String getRecipeContent(String Id) {
+    return DBUtils.blockOnFuture(DBUtils.recipe(Id).get()).getString(Recipe.CONTENT_KEY);
+  }
+
+  public RecipeMetadata getRecipeMetadata(String Id) {
+    return DBUtils.blockOnFuture(DBUtils.recipeMetadata(Id).get()).toObject(RecipeMetadata.class);
+  }
+
+  public String addRecipe(RecipeMetadata newRecipeMetadata, String newContent) {
```

Comment:
All methods here should have javadoc added, like in makeUserPropertyTrue.

Reply:
Small change; javadoc should be added to `DBInterface`, not `FirestoreDB`.

Reply:
okay, gotcha